os/net/routing/rpl-lite/rpl.c

```diff
@@ -163,7 +163,7 @@ rpl_set_prefix_from_addr(uip_ipaddr_t *addr, unsigned len, uint8_t flags)
   }
 
   /* Try and initialize prefix */
-  memset(&curr_instance.dag.prefix_info.prefix, 0, sizeof(rpl_prefix_t));
```

Comment:
I think it's the `sizeof` argument that needs to be changed, no the first argument of memset. The goal appears to be set the IPv6 address to zero, as the other members are initialized explicitly later in the code.

Reply:
this is code from `prl-classic/rpl-dag.c` :

```
int
rpl_set_prefix(rpl_dag_t *dag, uip_ipaddr_t *prefix, unsigned len)
{
  rpl_prefix_t last_prefix;
  uint8_t last_len = dag->prefix_info.length;

  if(len > 128) {
    return 0;
  }
  if(dag->prefix_info.length != 0) {
    memcpy(&last_prefix, &dag->prefix_info, sizeof(rpl_prefix_t));
  }
  memset(&dag->prefix_info.prefix, 0, sizeof(dag->prefix_info.prefix));    //<-  looks like rpl-lite need this line same?
  memcpy(&dag->prefix_info.prefix, prefix, (len + 7) / 8);

```

Reply:
yep, refixed it.